proto/iceflow/node-instance.proto

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2024 The IceFlow Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+syntax = "proto3";
+
+package iceflow;
+
+// Service definition for repartitioning node instances.
+service NodeInstance {
+  // Performs the repartition of a node instance.
+  rpc Repartition(RepartitionRequest) returns (RepartitionResponse) {}
+
+  // Retrieves production stats for a given edge from the node instance.
+  rpc QueryProductionStats(ProductionStatsRequest) returns (ProductionStatsResponse) {}
+
+  // Retrieves consumption stats for a given edge from the node instance.
+  rpc QueryConsumptionStats(ConsumptionStatsRequest) returns (ConsumptionStatsResponse) {}
```

Comment:
IMO it would be sufficient to just have a single RPC call that returns a list of all incoming and outgoing edges and production/consumption stats all at once.

AFAIK, there shouldn't be any reason why we would request those stats for only a single edge (at least for now).

Reply:
Good point, I will adjust the message definitions acordingly.

Reply:
I now moved the querying of all production and consumption stats into a single `QueryStats` RPC.